local/modules/TheliaSmarty/Template/Plugins/UrlGenerator.php

```diff
@@ -57,25 +78,39 @@ public function generateUrlFunction($params, &$smarty)
             );
         }
 
+        if ($routeId && $router) {
+            // get url by router and id
+            /** @var Router $router */
+            $router = $this->container->get($router);
```

Comment:
I think it would be better to have `$router = $this->container->get($routerId);`